### ISSUE TITLE
Morph + PowerSuit Flight Bug/Exploit

### ISSUE THREAD

**Issue by superkitten12** [xJon/The-1.12.2-Pack#214]
**Basic details**
1. Are you using a legitimate launcher and Minecraft account:
**yes**.
2. Are you using a computer with 8GB or more of RAM? ([How to check](https://www.wikihow.com/Check-Computer-RAM)):
**yes**.
3. Are you using Java 8, 64-bit? ([How to check](https://www.wikihow.com/Determine-Java-Version)):
**yes**.

**Information required**
1. Is the modpack crashing: **no**
2. Crash/latest log *link* : **n/a**
3. Is Optifine installed or any other additional mods: **yes**

**Describe the issue**

While transformed as any creature while wearing a Power Armor Chestplate (from MachineMuse's Modular Powersuits) that has been modified with the "Glider" upgrade, you gain the ability to have full flight. (Glider normally does not provide the ability of full flight and morphs have no powers.) Transforming back to a normal player gets rid of your flight, unless you are still flying, in which case you keep the ability until you are grounded. Removing the chestplate or it's upgrade while airborne will similarly not remove the ability to fly only when you are airborne when doing so. Updating game mode (both c->s ans s->c) will make you stop flying only when you are a player, aren't wearing the chestplate, or have removed the upgrade. If you still fulfill all of the requirements for the glitch when switching modes, nothing happens. Closing and reentering the world will make you stop flying temporarily if you still meet all requirements, but if you are missing one then you will also lose flight. All of this also applies to the Jetpack upgrade, but not the Parachute upgrade. This issue obviously makes such expensive late-game items from multiple mods which mostly only offer flight completely useless as you can then unlock flight very easily relatively early-mid game. For comparison, you can fly this way for just 28 iron, 6 osmium,7 redstone, 10 hardened glass, and an emerald(all of which is super easy to get in such low quantities). To compare it with I will use mystical agriculture as an example since it is easiest to calculate all factors and numbers. to unlock flight with this mod you need at least 32,736 inferium (not including what was used to get however many crops or the infusion crystals needed), as well as 24 wither skulls, 48 diamonds, 12 emeralds, and many other resources.
I have attached a screenshot to show me flying with nothing in my inventory. (after unmorphing and removing my armor).
![2020-02-15_18 59 11](https://user-images.githubusercontent.com/16422334/74597000-619e5580-5025-11ea-9baa-8888dae1cc60.png)

**Comment by xJon** [comment]
Was forwarded to the mod author. Will close now as there's nothing else we can do.
Thanks for reporting!

**Comment by superkitten12** [comment]
I know this is old but I wanted to give an update that I found other things that trigger this bug where you are put into a flying state. If you glide through a portal gun portal while morphed it will usually make you flying on the other end (assuming the morph was able to fit through the portal). If you just simply hold with jet boots long enough you will be made flying. Another one I noticed since i have them now too like the portal gun is if you go through a Viaduct while morphed and it would drop you off in the air you get flying. I tested all of these as magma slime since I'm usually using him to squeeze into tight spaces like between pipes and cables so these ones weren't tested any more than just what I've messed with in my survival world, and again most of the time unmorphing will keep the state.
